### PR TITLE
Allow explicit default with null value

### PR DESCRIPTION
**Pull request by stsewd**
This fix https://github.com/rtfd/readthedocs.org/issues/2813

- [x] Test case
- [x] Solution

### CODE REVIEW

**Comment by stsewd** [comment]
For now I'm only checking for the `requirements_file` default key, should I check for all defaults?

File: readthedocs_build/config/test_parser.py [hunk 1]
```diff
@@ -31,6 +31,18 @@ def test_parse_single_config():
...
+    config = parse(buf)
+    assert config[0]['base'] is None
+
+
+def test_parse_empty_value():
+    buf = StringIO(u'base:')
+    config = parse(buf)
+    assert config[0]['base'] is None
+
+
```
Comment (humitos) [inline readthedocs_build/config/test_parser.py]: I think a good addition here would be this test also:

```
def test_parse_empty_value():
    buf = StringIO(u'base: ""')
    config = parse(buf)
    assert config[0]['base'] is ''
```

File: readthedocs_build/config/test_config.py [hunk 2]
```diff
@@ -15,7 +15,6 @@
 from .config import PYTHON_INVALID
 from .validation import INVALID_BOOL
 from .validation import INVALID_CHOICE
-from .validation import INVALID_DIRECTORY
```
Comment (agjohnson) [inline readthedocs_build/config/test_config.py]: Let's keep this line for now, in case we have some side effects from removing a line like this.

Comment (stsewd) [reply readthedocs_build/config/test_config.py]: Done!

**Comment by stsewd** [comment]
@humitos I was thinking of that, but if look closely, always there are some additional checks before the second step, and at the end, we only will save 1 line of code per method.

**Comment by stsewd** [comment]
I forgot to delete one line on the merge :man_facepalming:

**Comment by agjohnson** [comment]
@stsewd shall we do a PR into rtd core for this as well?

**Comment by stsewd** [comment]
@agjohnson yeah!

**Comment by agjohnson** [comment]
Cool. Setting to blocking for now, we'll revisit this after.

**Comment by stsewd** [comment]
This is ported in https://github.com/rtfd/readthedocs.org/pull/4461

**Pull request closed without merging**